torch/_dynamo/variables/functions.py

```diff
@@ -30,16 +33,38 @@ def wrap_bound_arg(val, options):
         return variables.EnumVariable(val, **options)
     elif isinstance(val, (type, abc.ABCMeta)):
         return variables.UserDefinedClassVariable(val, **options)
+    elif istensor(val):
+        # Assumption: this is only called from InliningInstructionTranslator, in which case
+        # it is assured we will not 'reconstruct' this value in codegen, since the usage is
+        # entirely inlined.
+        # TODO - can we assert this? we expect to not reach this case for non-inline,
+        # since InstructionTranslator __init__ prepares VariableTrackers for args of top
+        # level function including defaults.
+        #
+        # TODO: wish i could assert this; but if the default tensor arg lives on
+        # a NNModuleVariable, currently the UserFunctionVariable itself has no source
+        # so we need to fix that first
+        # assert source is not None, "source must be provided for tensor arg"
+        if source:
+            return tx.output.register_attr_or_module(
+                val,
+                # should there be a "name" here?
+                source=source,
+            )
+        else:
+            return tx.output.register_attr_or_module(
+                val,
+            )
```

Comment:
Sorry, when is source None here?

Reply:
ok, this part should be resolved in the latest code.  I added a source to nnmodule call function, so that i could then build an appropriate source here.